### ISSUE TITLE
debug APi debug_traceTransaction negative gasCost

### ISSUE THREAD

**Issue by Savid** [NethermindEth/nethermind#6810]
**Description**
[debug_traceTransaction](https://docs.nethermind.io/interacting/json-rpc-ns/debug/#debug_tracetransaction) is giving negative `gasCost` values for `STOP` and `RETURN` opcodes.

**Steps to Reproduce**

Grab the transaction trace eg.
```bash
curl -X POST http://localhost:8545 \
  --data '{"jsonrpc":"2.0","method":"debug_traceTransaction","params":["0x48b7c61378bbd2fff08912109815161f5835735a0b6cbec9d0c7ef72b31619fb",{}],"id":1}' \
  -H "Content-Type: application/json"
```

Check for negative gas cost in the `structLogs` array

**Actual behavior**

Truncated `structLogs` response for random transaction i selected `0x48b7c61378bbd2fff08912109815161f5835735a0b6cbec9d0c7ef72b31619fb` on mainnet
```json
{
  "jsonrpc": "2.0",
  "result": {
    "gas": "0x335be",
    "failed": false,
    "returnValue": "0x",
    "structLogs": [
      {
        "pc": 98,
        "op": "RETURN",
        "gas": 240925,
        "gasCost": -3828,
        "depth": 4,
        "error": null,
        "stack": [
          "0x0",
          "0x20",
          "0x80"
        ],
        "storage": {}
      },
      {
        "pc": 367,
        "op": "RETURN",
        "gas": 185740,
        "gasCost": -3385,
        "depth": 4,
        "error": null,
        "stack": [
          "0x23b872dd",
          "0x20",
          "0x80"
        ],
        "storage": {
          "03c1b6c43b6549b150e056d5d4601ba0aededae00b30ca85227a4cd1b300c9d6": "03c1b6c43b6549b150e056d5d4601ba0aededae00b30ca85227a4cd1b300c9d6",
          "9b6883104bbe6b4f1d835989006f5f280cd909142f8d4f2576824cbb95d613d7": "9b6883104bbe6b4f1d835989006f5f280cd909142f8d4f2576824cbb95d613d7",
          "23677e5e13a8c743bcc9e13c4e141d9099e8b36dc925ad21b18df97876840a84": "23677e5e13a8c743bcc9e13c4e141d9099e8b36dc925ad21b18df97876840a84"
        }
      },
      {
        "pc": 600,
        "op": "STOP",
        "gas": 121586,
        "gasCost": -2689,
        "depth": 6,
        "error": null,
        "stack": [
          "0x22c0d9f"
        ],
        "storage": {
          "000000000000000000000000000000000000000000000000000000000000000c": "000000000000000000000000000000000000000000000000000000000000000c",
          "0000000000000000000000000000000000000000000000000000000000000009": "0000000000000000000000000000000000000000000000000000000000000009",
          "000000000000000000000000000000000000000000000000000000000000000a": "000000000000000000000000000000000000000000000000000000000000000a",
          "0000000000000000000000000000000000000000000000000000000000000008": "0000000000000000000000000000000000000000000000000000000000000008"
        }
      }
  ]
}
```

**Expected behavior**
Comparing to other execution layer clients, `gasCost` looks to be omitted.

**Desktop (please complete the following information):**
Please provide the following information regarding your setup:
 - Operating System: `Linux`
 - Version: `Nethermind/v1.25.4+20b10b35/linux-x64/dotnet8.0.2`

**Trace**

[0x48b7c61378bbd2fff08912109815161f5835735a0b6cbec9d0c7ef72b31619fb.json](https://github.com/NethermindEth/nethermind/files/14489195/0x48b7c61378bbd2fff08912109815161f5835735a0b6cbec9d0c7ef72b31619fb.json)

**Comment by LukaszRozmej** [comment]
Probably duplicate of #6810

**Comment by natebeauregard** [comment]
Closing this issue since it's a duplicate of #6669 and is fixed by #6831
